vmdb/app/views/layouts/_dhtmlxgrid.html.haml

```diff
@@ -19,7 +19,8 @@
   - unless @parent.nil?
     var miq_parent_id = "#{@parent.id}";
     var miq_parent_class = "#{@parent.class.base_class.to_s.underscore}";
-    if (miq_parent_class == "policy_set") miq_parent_class = "policy";
+    if (miq_parent_class == "policy_set") { miq_parent_class = "policy"; }
+    else if ((miq_parent_class == "ext_management_system") && ("#{@display}" == "hosts") && ("#{@layout}" == "ems_infra")) { miq_parent_class = "ems_infra" }
```

Comment:
@mzazrivec does this also fix sorting of any other items such as VMs, Datastores, Clusters  and does this fix sorting of sub-lists under other kinds of providers such as Cloud/Container.